scripts/cicd/lambda/lambda-publish.mjs

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env zx
+
+import { existsSync } from "fs"
+
+/**
+ *
+ * @param compiler tsc or ncc
+ */
+const compileWithYarn = async (compiler) => {
+  const useTSC = compiler === "tsc"
+  const useNCC = compiler === "ncc"
+  // clean deps
+  await $`rm -rf node_modules || :`
+
+  // need to compile ts
+  await $`yarn install`
+
+  // need to exclude node_modules in tsconfig.json
+  useTSC && (await $`yarn build`)
+  useNCC && (await $`yarn build-ncc`)
+
+  // clean deps
+  useTSC && (await $`rm -rf node_modules || :`)
+
+  // only install package in deps
+  useTSC && (await $`yarn install --prod`)
+
+  // clean package
+  await $`rm deployment_package.zip || :`
+
+  // zip package
+  // zip -q -r deployment_package.zip dist -x 'dist/*.mem'
+  useNCC && (await $`zip -q -r deployment_package.zip dist -x 'dist/*.mem'`)
+  useTSC && (await $`zip -q -r deployment_package.zip node_modules dist -x 'dist/*.map'`)
+}
```

Comment:
Hi @GeekEast 
Do you think the build process should be done in repo layer because we might have many different way to build, handle build process here might make this script complicated and it might not work for every repo

Reply:
this piece of code provides option to build with tsc or ncc.

So I put it in the cicd pipeline. 

Do you have an example for lambda function to build but different from this one?

Reply:
@GeekEast I don't have example but ncc is an example
Previously we use tsc to build for ts then we find ncc tool, so some repo using ncc. Meaning using tool to build before publish is more repo layer job imo

Reply:
make sense @tonynguyenit18 I will create a ticket for this task 

but will create a new scripts rather than replace the existing for backwards compatibility concern.

Reply:
https://predictivehire.atlassian.net/browse/ED-641

Reply:
Thanks @GeekEast 
Currently this script works fine to us, so please take your time